kyaml/yaml/compatibility.go

```diff
@@ -90,3 +93,58 @@ func IsValueNonString(value string) bool {
 }
 
 var stringType = reflect.TypeOf("string")
+
+// ValidValueForTag checks whether the given value is valid for the provided
+// yaml tag.
+func ValidValueForTag(value, tag string) (bool, error) {
```

Comment:
Can we use the OpenAPI validation libraries instead?

Reply:
I will take a look. But this issue isn't about schemas or validation according to a schema, but making sure we don't end up with a mismatch between the type and value of a yaml node.

Reply:
I think we can do this by using the go-yaml library to encode the value with a given type, and then see if the output includes the type marker (`!!<type>`). Will try it out.